### PR TITLE
LINBEE-9831 -Ask ai plugin

### PR DESCRIPTION
**Pull request by PavelLinearB**
Jira issue: LINBEE-9831
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=MmdXlJu)

image

### CODE REVIEW

File: plugins/filters/askAI/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2023 LinearB
```
Comment (vim-zz) [inline plugins/filters/askAI/LICENSE:3]:
```suggestion
Copyright (c) 2024 LinearB
```

File: plugins/filters/askAI/askAI.cm [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+automations:
+  generate_pr_desc:
+    if:
+      - true
+    run:
+      - action: add-comment@v1
+        args:
+          comment: |
+            {{ code | askAI("Based on the given context, search for new functions without tests and suggest the tests to add. If all functions are covered completely, return 'no tests to suggest.'", env.OPEN_AI_TOKEN) | encode }}
```
Comment (vim-zz) [inline plugins/filters/askAI/askAI.cm:14]: What's `code`? is it a new context? where is is defined?

Comment (vim-zz) [reply plugins/filters/askAI/askAI.cm:14]: Same for `encode` - is it documented?

File: plugins/filters/askAI/askAI.cm [hunk 3]
```diff
@@ -0,0 +1,14 @@
+triggers:
+  exclude:
+    branch:
+      - r/dependabot/
+
+automations:
+  generate_pr_desc:
+    if:
+      - true
```
Comment (MishaKav) [inline plugins/filters/askAI/askAI.cm:9]: maybe it should be only for pr_created and on commit?

Comment (vim-zz) [reply plugins/filters/askAI/askAI.cm:9]: another option is on `pr_created` and then only if you ask for re-evaluation in a comment (e.g. add comment like `/askai`)

File: plugins/filters/askAI/askAI.cm (hunk 2 above)
Comment (PavelLinearB) [reply plugins/filters/askAI/askAI.cm]: Adding the `encode` and `decode` documentation on a separate PR

**Review by vim-zz — APPROVED** [review]

File: plugins/filters/askAI/askAI.cm [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      - action: add-comment@v1
+        args:
+          comment: |
+            {{ source | askAI("Based on the given context, search for new functions without tests and suggest the tests to add. If all functions are covered completely, return 'no tests to suggest.'", env.OPEN_AI_TOKEN) | encode }}
+
+  generate_pr_desc_on_ask_ai_label:
+    on:
+      - label_added
+    if:
+      - {{ pr.labels | match(term="ask-ai-qa") | some }}
```
Comment (vim-zz) [inline plugins/filters/askAI/askAI.cm:22]:
```suggestion
      - {{ pr.labels | match(term="/ask-ai qa") | some }}
```

**Pull request merged**